packages/smooth_app/lib/widgets/smooth_app_bar.dart

```diff
@@ -0,0 +1,214 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// A custom [AppBar] with an action mode.
+/// If [action mode] is true, please provide at least an [actionModeTitle].
+class SmoothAppBar extends StatelessWidget implements PreferredSizeWidget {
+  SmoothAppBar({
+    this.leading,
+    this.automaticallyImplyLeading = true,
+    this.actionModeTitle,
+    this.actionModeSubTitle,
+    this.title,
+    this.actionModeActions,
+    this.actions,
+    this.flexibleSpace,
+    this.bottom,
+    this.elevation,
+    this.scrolledUnderElevation,
+    this.shadowColor,
+    this.surfaceTintColor,
+    this.backgroundColor,
+    this.foregroundColor,
+    this.iconTheme,
+    this.actionsIconTheme,
+    this.primary = true,
+    this.centerTitle,
+    this.excludeHeaderSemantics = false,
+    this.titleSpacing,
+    this.shape,
+    this.toolbarOpacity = 1.0,
+    this.bottomOpacity = 1.0,
+    this.toolbarHeight,
+    this.leadingWidth,
+    this.toolbarTextStyle,
+    this.titleTextStyle,
+    this.systemOverlayStyle,
+    this.actionMode = false,
+    this.actionModeCloseTooltip,
+    this.onLeaveActionMode,
+    Key? key,
+  })  : preferredSize =
+            _PreferredAppBarSize(toolbarHeight, bottom?.preferredSize.height),
+        super(key: key);
```

Comment:
> If [action mode] is true, please provide at least an [actionModeTitle].

Is it possible to add an assert for the condition specified?

Reply:
That's added 😇